iceoryx_binding_c/include/iceoryx_binding_c/chunk.h

```diff
@@ -17,19 +17,53 @@
 #ifndef IOX_BINDING_C_CHUNK_H
 #define IOX_BINDING_C_CHUNK_H
 
-#include "iceoryx_binding_c/internal/c2cpp_binding.h"
-
-/// @brief handle of the chunk header
-typedef void* iox_chunk_header_t;
+#include "iceoryx_binding_c/types.h"
 
 /// @brief gets the user-payload from the chunk-header
 /// @param[in] chunkHeader pointer to the chunk-header
 /// @return pointer to the user-payload
-void* iox_chunk_header_to_user_payload(iox_chunk_header_t const chunkHeader);
+void* iox_chunk_header_to_user_payload(iox_chunk_header_t* const chunkHeader);
+
+/// @brief gets the const user-payload from the const chunk-header
+/// @param[in] chunkHeader const pointer to the chunk-header
+/// @return const pointer to the user-payload
+const void* iox_chunk_header_to_user_payload_const(const iox_chunk_header_t* const chunkHeader);
+
+/// ==========================================================================================
+/// user-header option 1
+
+/// @brief gets the user-header from the chunk-header
+/// @param[in] chunkHeader pointer to the chunk-header
+/// @return pointer to the user-header
+void* iox_chunk_header_to_user_header(iox_chunk_header_t* const chunkHeader);
+
+/// @brief gets the const user-payload from the const chunk-header
+/// @param[in] chunkHeader const pointer to the chunk-header
+/// @return const pointer to the user-header
+const void* iox_chunk_header_to_user_header_const(const iox_chunk_header_t* const chunkHeader);
+
+/// user-header option 2
+
+/// @brief gets the user-header from the user-payload
+/// @param[in] userPayload pointer to the user-payload
+/// @return pointer to the user-header
+void* iox_chunk_user_header_from_user_payload(void* const userPayload);
+
+/// @brief gets the const user-header from the const user-payload
+/// @param[in] userPayload const pointer to the user-payload
+/// @return const pointer to the user-header
+const void* iox_chunk_user_header_from_user_payload_const(const void* const userPayload);
```

Comment:
The big question is which API do we want?

I'm leaning towards option 1, since it's similar what we have in our untyped C++ API where one has to get the pointer to the `ChunkhHeader` first in order to get the user-header, e.g. `ChunkHeader::fromUserPayload(p)->userHeader<Foo>()`.
With option one this would become `
void* foo = iox_chunk_header_to_user_header(iox_chunk_header_from_user_payload(p));`
It's also slightly more type safe than option 2 because one has to supply a `iox_chunk_header_t` pointer. Option 2 would be shorter though but one can write a small helper function for that if needed

Reply:
+1 for option 1

Reply:
so option 1 is choosen